pkg/discovery/server/server.go

```diff
@@ -44,6 +46,7 @@ func NewServer(pdControl pdapi.PDControlInterface, cli versioned.Interface, kube
 func (s *server) registerHandlers() {
 	ws := new(restful.WebService)
 	ws.Route(ws.GET("/new/{advertise-peer-url}").To(s.newHandler))
+	ws.Route(ws.GET("/newdm/{advertise-peer-url}").To(s.newDMHandler))
```

Comment:
is it possible to add a parameter to `/new/{advertise-peer-url}` API for dm? e.g. `service=dm`

`newHandler` and `newDMHandler` are similar, we can avoid duplicate code if we can.

Reply:
addressed in https://github.com/pingcap/tidb-operator/pull/3098/commits/609e797ddf5cdeff1eec66c6fef0804ec9c2b198